movement/movement.c

```diff
@@ -691,3 +713,81 @@ void cb_tick(void) {
         movement_state.subsecond++;
     }
 }
+
+/** Figures out the next scheduled event for the HPT to wake up for.
+ * Must be called whenever:
+ * - a new event is scheduled
+ * - an old event is deleted
+ * - an event is triggered
+ * - the timer overflows
+ */
+static void _movement_hpt_schedule_next_event() {
+    uint64_t next = UINT64_MAX;
+    for(uint8_t req_idx = 0; req_idx < MOVEMENT_NUM_FACES; ++req_idx) {
+        uint64_t event = hpt_scheduled_events[req_idx];
+        if(event < next) {
+            event = next;
+        }
+    }
+
+    // if an event is scheduled and the timer is currently tracking this overflow cycle
+    if(next != UINT64_MAX && (next >> 32) == hpt_overflows) {
+        watch_hpt_schedule_callback((uint32_t)(next & UINT32_MAX));
+    } else {
+        watch_hpt_disable_scheduled_callback();
+    }
+}
+
+void movement_hpt_request(void) {
+    movement_hpt_request_face(movement_state.current_face_idx);
+}
+void movement_hpt_request_face(uint8_t face_idx) {
+    HPT_REQUESTS_T old_hpt_requests = hpt_enable_requests;
+    SET_BIT(hpt_enable_requests, face_idx);
+    if(old_hpt_requests == 0) {
+        watch_hpt_enable();
+    }
+}
+
+void movement_hpt_relenquish(void) {
+    movement_hpt_relenquish_face(movement_state.current_face_idx);
+}
+void movement_hpt_relenquish_face(uint8_t face_idx) {
```

Comment:
Small correction: `relinquish`.

Reply:
I thought I spelled that wrong. I'm gonna rename it to "release" anyway, because "relinquish" is too fancy.